jerry-core/ecma/operations/ecma-get-put-value.c

```diff
@@ -52,7 +52,22 @@ ecma_op_get_value_lex_env_base (ecma_object_t *ref_base_lex_env_p, /**< referenc
   /* 3. */
   if (unlikely (is_unresolvable_reference))
   {
-    return ecma_raise_reference_error (ECMA_ERR_MSG ("Cannot resolve reference."));
+#ifdef JERRY_ENABLE_ERROR_MESSAGES
+    ecma_ref_ecma_string (var_name_string_p);
+    ecma_string_t *string_array[] =
+    {
+      var_name_string_p,
+      ecma_get_magic_string (LIT_MAGIC_STRING_SPACE_CHAR),
+      ecma_get_magic_string (LIT_MAGIC_STRING_ERROR_IS_NOT_DEFINED),
+    };
+
+    ecma_string_t *msg_str_p = ecma_join_ecma_strings (string_array, JERRY_NUM_OF_ELEMENTS (string_array));
+    ECMA_DEFINE_ERROR_VALUE (error_value, ECMA_ERROR_REFERENCE, msg_str_p);
+    ecma_deref_ecma_string (msg_str_p);
```

Comment:
If we use this utility solely for creating errors, we could create the error object in the utility to reduce binary size.